orangecontrib/text/widgets/owconcordance.py

```diff
@@ -260,16 +264,14 @@ def update_widget(self):
         self.conc_view.resizeRowsToContents()
 
         if self.corpus is not None:
-            self.n_documents = len(self.corpus)
             self.n_matching = '{}/{}'.format(
                 self.model.matching_docs() if self.word else 0,
-                self.n_documents)
-            self.n_tokens = sum(map(len, self.corpus.tokens)) \
-                if self.corpus.has_tokens() else 'n/a'
+                len(self.corpus))
+            self.n_tokens = sum(map(len, self.model.tokens)) \
+                if self.model.tokens is not None else 'n/a'
             self.n_types = len(self.corpus.dictionary) \
```

Comment:
We should probably add something like `self.n_types = len(set(tokens))` to ConcordanceModel's set_tokens method and use that instead.